core/jvm/src/LocalDate.kt

```diff
@@ -52,6 +52,11 @@ public actual class LocalDate internal constructor(
         @Suppress("FunctionName")
         public actual fun Format(block: DateTimeFormatBuilder.WithDate.() -> Unit): DateTimeFormat<LocalDate> =
             LocalDateFormat.build(block)
+
+        // even though this class uses writeReplace (so serialVersionUID is not needed for a stable serialized form), a
+        // stable serialVersionUID means exceptions caused by deserialization of malicious streams will be consistent
+        // (InvalidClassException vs. InvalidObjectException, see MaliciousJvmSerializationTest)
```

Comment:
Is this really needed? As soon as some runtime serialization exception is thrown on a malicious stream, does it really matter which one?

Reply:
I realized that explicitly declaring the `serialVersionUID` is needed to properly do the tests in `MaliciousJvmSerializationTest`. The malicious serial streams in that test are designed to bypass the proxy given no defensive `readObject` method. But that only works if they have the correct `serialVersionUID` in them. So they would have to be regenerated each time the `serialVersionUID` changes (which e.g. happened for `LocalDate` in d15ec21e1d53944168357bf70934fec46107730f due to new methods in that class, it is now `-9141474295923275006L`). Otherwise we would just test that `ObjectInputStream` detects and rejects invalid `serialVersionUID`s.